pkg/proc/breakpoints.go

```diff
@@ -70,10 +72,17 @@ type Breakpoint struct {
 	Cond ast.Expr
 	// internalCond is the same as Cond but used for the condition of internal breakpoints
 	internalCond ast.Expr
+	// HitCond: if not nil the breakpoint will be triggered only if the evaluated HitCond returns
+	// true with the TotalHitCount.
+	HitCond *ast.BinaryExpr
 
 	// ReturnInfo describes how to collect return variables when this
 	// breakpoint is hit as a return breakpoint.
 	returnInfo *returnBreakpointInfo
+
+	// RequestString is a string used by the debug adapter to
+	// identify a breakpoint with a particular request.
+	RequestString string
```

Comment:
The name field has stricter rules about what characters are allowed. So to include the file path, we would have to sanitize the string somehow.

Reply:
It would be reasonable to move those validity checks for the name out of debugger and into rpc2/server.go and rpc1/server.go

Reply:
Ok sounds good. Will do that.

Reply:
Seems odd and not very intuitive to have different validity methods for the name field just depending on which API you're using. Maybe we relax the name check so that this new format is allowed as well while still validating the input?